lib/order.rb

```diff
@@ -0,0 +1,72 @@
+#require_relative '..lib/customer'
+class Order
+  @@orders = []
+
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status=:pending)
+    @id = id
+    # if @id.class != integer
+    #   puts "not a num"
+    # end
+
+    @products = products
+    # if @product_cost.class != hash
+    #   puts "not a hash"
+    # end
+
+    @customer = customer
+
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError unless
+    @fulfillment_status == :pending || @fulfillment_status == :paid || @fulfillment_status == :processing || @fulfillment_status == :shipped || @fulfillment_status == :complete
+  end
+
+  def add_product(name, price)
+    if @products.include?(name)
+      raise ArgumentError
+    end
+    return @products[name] = price
```

Comment:
This line does a lot of things at once, which is really cool, but I want to call out the steps since it's a little obscure:

It adds a new entry into the `@products` hash, with the key/value pair of the value of `name` as the key and the value of `price` as the value...
And it returns the evaluation of `@products[name] = price`, which Ruby says is going to be `price`.

If that doesn't feel very obvious to you (it isn't obvious to me), then I would recommend breaking it up as:
```ruby
@products[name] = price
return price
```